pkg/execution/executor/executor.go

```diff
@@ -891,14 +891,13 @@ func (e *executor) HandleResponse(ctx context.Context, i *runInstance, resp *sta
 		// Check if this step permanently failed.  If so, the function is a failure.
 		if !resp.Retryable() {
 			// TODO: Refactor state input
-			if performedFinalization, err := e.finalize(ctx, i.md, i.events, i.f.GetSlug(), *resp); err != nil {
+			if _, err := e.finalize(ctx, i.md, i.events, i.f.GetSlug(), *resp); err != nil {
```

Comment:
Should we refactor finalize to remove performedDeletion altogether, since it's unused and not quite trustworthy?